index.js

```diff
@@ -0,0 +1,213 @@
+    searchText = event.target.value;
+});
+
+//Listener for search submit
+form.addEventListener('submit', event => {
+    event.preventDefault();
+    modeLatestNews = false;
+    topic = searchText;
+    form.reset(); 
+    searchText = '';
+    document.querySelector(".search-box").style.display = "none";
+    APIQuery = `https://newsapi.org/v2/everything?language=en&page=${page}&pageSize=${pageSize}&q=${topic}&apiKey=4fede2e79736471581f9259c131f0fc2`;
+    document.querySelector('.page-nav').setAttribute('style','display: flex');
+    fetchArticles();
+    pageTitle.innerHTML = 'Search results for ' + topic;
+});
+
+//Listener for next page button
+pageNextButton.addEventListener('click', event => {
+    page++;
+    APIQuery = `https://newsapi.org/v2/everything?language=en&page=${page}&pageSize=${pageSize}&q=${topic}&apiKey=4fede2e79736471581f9259c131f0fc2`;
+    fetchArticles();
+});
+
+//Listener for previous page button
+pagePrevButton.addEventListener('click', event => {
+    if (page > 1) {
+        page--;
+        APIQuery = `https://newsapi.org/v2/everything?language=en&page=${page}&pageSize=${pageSize}&q=${topic}&apiKey=4fede2e79736471581f9259c131f0fc2`;
+        fetchArticles();
+    } 
+});
+
+//Listener for toggle
+toggle.addEventListener('click', function() {
+    toggleNumber = !toggleNumber;
+    if (toggleNumber) {
+        toggleContainer.style.clipPath = 'inset(0 0 0 50%)';
+        toggleContainer.style.backgroundColor = '#D74046';
+        country = 'us';
+    } else {
+        toggleContainer.style.clipPath = 'inset(0 50% 0 0)';
+        toggleContainer.style.backgroundColor = 'dodgerblue';
+        country = 'gb';
+    }
+    APIQuery = `https://newsapi.org/v2/top-headlines?country=${country}&category=${category}&apiKey=4fede2e79736471581f9259c131f0fc2`;  
+    modeLatestNews = true;
+    document.querySelector('.page-nav').setAttribute('style','display: none');
+    fetchArticles();
+    setTimeout(closeNav,500);
+});
+
+//Listener for Newsroom link
+logo.addEventListener('click', event => {
+    topic = 'Politics';
+    page = 1;
+    APIQuery = `https://newsapi.org/v2/everything?language=en&page=${page}&pageSize=${pageSize}&q=${topic}&apiKey=4fede2e79736471581f9259c131f0fc2`;
+    modeLatestNews = false;
+    document.querySelector('.page-nav').setAttribute('style','display: flex');
+    pageTitle.innerHTML = topic;
+    fetchArticles();
+});
+
+//Fetch data
+function fetchArticles() {
+    console.log('fetching articles now...');
+    console.log(APIQuery);
+    fetch(APIQuery)
+    .then(response => {
+        if (!response.ok) throw response;
+        return response.json();})
+    .then(body => renderArticles(body))
+    .catch(error => console.log(error));
+}
+
+//Function to generate HTML elements for articles    
+function renderArticles(body) {
+    articles.innerHTML = '';
+    body.articles.forEach(item => {
+        const article = document.createElement('article');
```

Comment:
I would suggest extracting display logic to own function. Also, it might be simpler to generate content as HTML string using string templates

Reply:
thx, changed to string literals... much shorter now!